### DISCUSSION TITLE
Should we use GCP storage buckets

### DISCUSSION THREAD

**Discussion by cbf66** [arXiv/discussion#3] [General]
See [https://github.com/arXiv/arxiv-canonical/tree/develop/schema-extended](url)

Pros:

- Inexpensive
- Well supported

Cons:

- Not as "natural" as filesystems
- Shouldn't use a GCP proprietary feature (although all the Cloud services have a comparable feature)

**Comment by bdc34** [comment]
On the con of "not as natural", it can be eliminated with a library like https://pypi.org/project/smart-open/

_smart_open is a Python 3 library for efficient streaming of very large files from/to storages such as S3, GCS, Azure Blob Storage, HDFS, WebHDFS, HTTP, HTTPS, SFTP, or local filesystem. It supports transparent, on-the-fly (de-)compression for a variety of different formats._

_smart_open is a drop-in replacement for Python’s built-in open(): it can do anything open can (100% compatible, falls back to native open wherever possible), plus lots of nifty extra stuff on top._

**Comment by bmaltzan** [comment]
Another pro, object stores have a higher/no-limit on the number of objects in a bucket.
